src/main/java/com/microsoft/graph/http/CoreHttpProvider.java

```diff
@@ -374,11 +387,11 @@ public Request authenticateRequest(Request request) {
 									.newBuilder()
 									.connectTimeout(connectionConfig.getConnectTimeout(), TimeUnit.MILLISECONDS)
 									.readTimeout(connectionConfig.getReadTimeout(), TimeUnit.MILLISECONDS)
-									.followRedirects(false)
+									.followRedirects(false) //TODO those are duplicated for the time being with core, remove in 3.0
 									.protocols(Arrays.asList(Protocol.HTTP_1_1)) //https://stackoverflow.com/questions/62031298/sockettimeout-on-java-11-but-not-on-java-8
 									.build();
 			}
-			if (authenticationProvider != null) {
+			if (authenticationProvider != null) { //TODO remove the authentication provider as this is done by the interceptor anyway
```

Comment:
Same here: link to corresponding github issue.

To save time, I like @MIchaelMainer's tactic of adding the title (with an optional quick summary) for creating the link as a reference quickly. The details can be filled afterwards.

Reply:
done #517, same question about the comment

Reply:
```suggestion
			if (authenticationProvider != null) { //TODO https://github.com/microsoftgraph/msgraph-sdk-java/issues/517
```